include/deal.II/lac/parpack_solver.h

```diff
@@ -690,6 +689,16 @@ void PArpackSolver<VectorType>::solve
       break;
     }
 
+  //Check for possible options for symmetric problems
+  Assert(!additional_data.symmetric || strcmp(which, "LR"),
+         ExcMessage("'largest real part' can only be used for non-symmetric problems!"));
+  Assert(!additional_data.symmetric || strcmp(which, "SR"),
+         ExcMessage("'smallest real part' can only be used for non-symmetric problems!"));
+  Assert(!additional_data.symmetric || strcmp(which, "LI"),
+         ExcMessage("'largest imaginary part' can only be used for non-symmetric problems!"));
+  Assert(!additional_data.symmetric || strcmp(which, "SI"),
+         ExcMessage("'smallest imaginary part' can only be used for non-symmetric problems!"));
```

Comment:
A trivial complaint, but we should use `std::strcmp` here.

I think it would be nicer to just use `std::string` in this function but that was already there.